tests/50federation/34room-backfill.pl

```diff
@@ -65,7 +66,16 @@
          )->then( sub {
             my ( $body ) = @_;
 
-            my $room_id = $body->{room_id};
+            $room_id = $body->{room_id};
+
+            # wait for it to arrive
+            await_sync_timeline_contains(
+               $user, $room_id,
+               check => sub {
+                  $_[0]->{type} eq "m.room.member"
+               },
+            );
```

Comment:
isn't this the point of the next bit happening in a loop? the comment even says so?

and if we need to make this change, can we remove the loop? Or at least update the comments?

Reply:
I'll remove the loop as it shouldn't be needed anymore.

Reply:
Done.

Reply:
I'm moderately interested to know why the loop was insufficent, but :man_shrugging:

Reply:
The way this test and `matrix_get_room_messages` handles tokens breaks things. This function:
 - Hits `/sync`.
 - If the `from` parameter was set, uses it to call `/messages`, else uses `next_batch`. 

We pass an empty `from` at first, resulting in `next_batch` being used as the `?from=` param for `/messages`. The test then sets `from` to be `chunk.end` to backpaginate more results.

If the membership event has come down `/sync` then `next_batch` represents a point after the event has been sent, so using that token with `/messages` will return that event again which the test wants. If the membership event has not come down `/sync` yet then `next_batch` represents a point before the membership event, so using that token with `/messages` will not return the membership event, causing the test to fail.

The test assumes that `/sync` will have the membership event, but that's not always true until now.